src/utils.ts

```diff
@@ -87,3 +88,29 @@ export function showDataDirectory(appId: string) {
         void vscode.env.openExternal(vscode.Uri.file(dataDirectory))
     }
 }
+
+/**
+ * Finds the directory of the file with `fileName` in the given directory and return
+ * the first match.
+ * @param dir The directory to search
+ * @param fileName The file name to search for
+ */
+export function findFileParent(dir: PathLike, fileName: string): Promise<PathLike | null> {
+    return new Promise((resolve) => {
+        const walker = walk(dir.toString(), {
+            followLinks: false,
```

Comment:
Not sure if this is what we want

Reply:
I guess let us leave some comment there for now and we can change it if someone complains ;)